src/iuse.cpp

```diff
@@ -2619,6 +2621,77 @@ cata::optional<int> iuse::noise_emitter_on( Character *p, item *it, bool t, cons
     return it->type->charges_to_use();
 }
 
+cata::optional<int> iuse::emf_passive_off( Character *p, item *it, bool, const tripoint & )
+{
+    if( !it->ammo_sufficient( p ) ) {
+        p->add_msg_if_player( _( "It's dead." ) );
+        return cata::nullopt;
+    } else {
+        p->add_msg_if_player( _( "You turn the emf detector on." ) );
+        it->convert( itype_emf_detector_on ).active = true;
+    }
+    return it->type->charges_to_use();
+}
+
+cata::optional<int> iuse::emf_passive_on( Character *p, item *it, bool t, const tripoint &pos )
+{
+    if( t ) { // Normal use
+        // need to calculate distance to closest thing electrical thing
+
+        // set distance as farther than the radius
+        const int max = 10;
+        int distance = max + 1;
+
+        creature_tracker &creatures = get_creature_tracker();
+        map &here = get_map();
+        // can't get a reading during a portal storm
+        if( get_weather().weather_id == weather_type_id( "portal_storm" ) ) {
+            sounds::sound( pos, 6, sounds::sound_t::alarm, _( "BEEEEE-CHHHHHHH-eeEEEEEEE-CHHHHHHHHHHHH" ), true,
+                           "tool", "emf_detector" );
+            // skip continuing to check for locations
+            return it->type->charges_to_use();
+        }
+
+        for( const auto &loc : closest_points_first( pos, max ) ) {
+            const Creature *critter = creatures.creature_at( loc );
+
+            // if the creature exists and is either a robot or electric
+            bool found = critter != nullptr && critter->is_electrical();
+
+            // check for an electrical field
+            if( !found ) {
+                for( const auto &fd : here.field_at( loc ) ) {
+                    if( fd.first->has_elec ) {
+                        found = true;
+                        break;
+                    }
+                }
+            }
+
+            // if an electrical field or creature is nearby
+            if( found ) {
+                distance = rl_dist( pos, loc );
+                if( distance <= 3 ) {
+                    sounds::sound( pos, 4, sounds::sound_t::alarm, _( "BEEEEEEP BEEEEEEP" ), true, "tool",
+                                   "emf_detector" );
+                } else if( distance <= 7 ) {
+                    sounds::sound( pos, 2, sounds::sound_t::alarm, _( "BEEP BEEP" ), true, "tool",
```

Comment:
```suggestion
                    sounds::sound( pos, 3, sounds::sound_t::alarm, _( "BEEP BEEP" ), true, "tool",
```

To differentiate it from case of farthest distance.